code/modules/shuttle/on_move.dm

```diff
@@ -11,6 +11,28 @@
 		return 0
 	. = ..()
 
+/obj/machinery/atmospherics/onShuttleMove()
+	. = ..()
+	for(DEVICE_TYPE_LOOP)
+		if(get_area(nodes[I]) != get_area(src))
+			nullifyNode(I)
+
+/obj/structure/cable/onShuttleMove()
+	. = ..()
+	var/A = get_area(src)
+	//cut cables on the edge
+#define DIR_CHECK_TURF_AREA(X) (get_ranged_target_turf(src, X, 1) != A)
+	if(DIR_CHECK_TURF_AREA(NORTH) || DIR_CHECK_TURF_AREA(SOUTH) || DIR_CHECK_TURF_AREA(EAST) || DIR_CHECK_TURF_AREA(WEST))
+		cut_cable_from_powernet()
+
+/obj/structure/disposalpipe/onShuttleMove()
+	. = ..()
+	var/A = get_area(src)
+	//break pipes on the edge
+	if(DIR_CHECK_TURF_AREA(NORTH) || DIR_CHECK_TURF_AREA(SOUTH) || DIR_CHECK_TURF_AREA(EAST) || DIR_CHECK_TURF_AREA(WEST))
+		deconstruct(FALSE)
```

Comment:
I'm also pretty sure that disposals pipes don't form a network and just pass the holder along to the next pipe in line if there is one at all so the bug doesn't actually exist.

Counterpoint 1: The singularity ripping pipes up doesn't break pipes near those pipes, nor does this pipe moving break the pipe it was attached to
Counterpoint 2: That's not an argument, that's a statement!